docker/filebeat/filebeat.yml

```diff
@@ -25,150 +25,201 @@ filebeat.autodiscover:
             contains:
-              docker.container.image: "apm-server"
+              docker.container.image: "opbeans-node"
           config:
             - type: container
               paths:
-                - '/var/lib/docker/containers/*/*.log'
+                - "/var/lib/docker/containers/*/${data.docker.container.id}-json.log"
               containers.ids:
                 - "${data.docker.container.id}"
-              fields_under_root: true
-              json.keys_under_root: true
-              json.overwrite_keys: true
-              json.add_error_key: true
-              json.message_key: message
+              multiline.pattern: '^ '
+              multiline.negate: false
+              multiline.match: after
         - condition:
             contains:
-              docker.container.image: "filebeat"
+                docker.container.image: "opbeans-dotnet"
           config:
             - type: container
               paths:
-                - '/var/lib/docker/containers/*/*.log'
+                - "/var/lib/docker/containers/*/${data.docker.container.id}-json.log"
               containers.ids:
                 - "${data.docker.container.id}"
+              tail_files: true
               fields_under_root: true
               json.keys_under_root: true
               json.overwrite_keys: true
               json.add_error_key: true
               json.message_key: message
+              tags: ["${docker.container.labels.com_docker_compose_service}"]
         - condition:
             contains:
-              docker.container.image: "heartbeat"
+                docker.container.image: "opbeans-go"
           config:
             - type: container
               paths:
-                - '/var/lib/docker/containers/*/*.log'
+                - "/var/lib/docker/containers/*/${data.docker.container.id}-json.log"
               containers.ids:
                 - "${data.docker.container.id}"
+              tail_files: true
               fields_under_root: true
               json.keys_under_root: true
               json.overwrite_keys: true
               json.add_error_key: true
               json.message_key: message
+              tags: ["${docker.container.labels.com_docker_compose_service}"]
         - condition:
             contains:
-              docker.container.image: "kibana"
+                docker.container.image: "opbeans-java"
           config:
             - type: container
               paths:
-                - '/var/lib/docker/containers/*/*.log'
+                - "/var/lib/docker/containers/*/${data.docker.container.id}-json.log"
               containers.ids:
                 - "${data.docker.container.id}"
+              tail_files: true
               fields_under_root: true
               json.keys_under_root: true
               json.overwrite_keys: true
               json.add_error_key: true
               json.message_key: message
+              include_lines: ["^{"]
```

Comment:
Doesn't this simply drop log lines that aren't JSON?

We have a similar issue with opbeans-python, and there we (as in, @basepi) went with a conditional:

https://github.com/elastic/apm-integration-testing/blob/971186146810d008794a158240f5329954460287/docker/filebeat/filebeat.yml#L129-L134

(I see you removed that conditional as well in this PR, is that intentional?)

Reply:
define this processor is the same that include only JSON lines, all the rest of lines are ignored because there is no alternative definition to process them.

>(I see you removed that conditional as well in this PR, is that intentional?)

yes, I did no see any issues with the Python log so I've removed it, we can include also the include setting to avoid issues.

Reply:
When using the current master, I see a lot of log lines that don't match that regex in the processor, e.g.

    [2019-10-21 11:55:27,123: INFO/MainProcess] Received task: opbeans.tasks.update_stats[bdadda02-95a8-4f93-a1c5-47a6b93f52ae] 

So it doesn't look to me as if they are getting dropped.

I don't think we should drop _any_ log lines. While it's not really mission critical for ourselves, users will see this filebeat.yml in google searches etc. and use it as starting point for their own setups. We should avoid potential footguns like "drop anything that doesn't look like JSON".

Reply:
agree, but with your filter, we lose any JSON field that is not in the message field and the original message field

i.e. date and level will be lost

```
{ date: "...", level: "...", message: "..."}
```

I am going to rethink the thing to try to keep it as simple as possible without lose anything, we do not want to maintain 10 different ways to process the logs

Reply:
I've realized that the Python Agent does not have a pure JSON output this is the most similar message toa JSON, you should preprocess it by removing the first part of the message before to parse the JSON

```
[2019-10-21 16:22:12,044: DEBUG/MainProcess] {'event': 'flushing due to time since last flush 10.544s > max_flush_time 10.514s', '@timestamp': '2019-10-21T16:22:12.044753Z', 'logger': 'elasticapm.transport', 'level': 'debug'}
```

Reply:
Yes, we're aware of that, that's why we introduced that processor I linked to above. Unfortunately, it's not an easy fix, as this logging format is set up by Celery (the background task manager). We spent some time in trying to fix it, but didn't find a good solution yet. Hence we'd really prefer if the processor could be kept in place for now.

Reply:
I was wrong there are some JSON lines in Python logs